### ISSUE TITLE
[Dialogs] Non-dismissable (modal) Alerts can be dismissed with VoiceOver "Z" gesture

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#4259]
## Description
When showing a non-dismissable Alert, tapping outside the alert will not dismiss it. However, using the 2-finger "Z" gesture with VoiceOver *will* dismiss the alert.  I think either VO Z-gesture **should not** work or tapping outside the alert **should** work.

## Reproduction steps
1. Turn off VoiceOver
1. Open the catalog
1. Open Dialogs &rarr; AlertController
1. Select "Non-dimissable alert"
1. Try to dismiss the alert by tapping outside the alert view
1. Repeat with VoiceOver on and using the 2-finger "Z" gesture to dismiss (instead of tapping outside)

## Expected behavior
Both paths, VoiceOver or tapping, result in consistent behavior.

## Actual behavior
VoiceOver permits dismissing the Alert without selecting one of the available actions. Without VoiceOver, users must tap one of the available actions.

## Platform (please complete the following information)
iPhone 5s/iOS 11.2.6

**Comment by ianegordon** [comment]
We need insight from the A11y team.

**Comment by ianegordon** [comment]
Seems reasonable, and I share your concerns about landed in an unexpected state. I've reached out to the A11y team in case they have any insight / recommendations.

Additional points:
UIAlertController does *not* allow the escape gesture.
We can explicitly invoke the escape action programmatically, so we may be able to vary behavior based on MDCDialogPresentationController's existing dismissOnBackgroundTap property.
